app-template/src/CookiesNotification.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react'
+import { withRouter, RouteComponentProps } from 'react-router-dom'
+import { withCookies, ReactCookieProps } from 'react-cookie'
+
+export type CookiesNotificationToken = {
+  hasAgreedToCookies: boolean
+}
+
+type Props = RouteComponentProps & ReactCookieProps
+
+const hasAgreedToCookiesKey = 'org.sagebionetworks.security.cookies.portal.notification.okclicked'
+
+class CookiesNotification extends React.Component<Props, CookiesNotificationToken> {
+
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      hasAgreedToCookies: true
+    }
+    this.setHasAgreedToCookies = this.setHasAgreedToCookies.bind(this)
+  }
+
+  componentDidMount() {
+    const okClicked = this.props.cookies!.get(hasAgreedToCookiesKey)
+    if (!okClicked) {
+      this.setState({
+        hasAgreedToCookies: false
+      })
+    }
+  }
+
+  setHasAgreedToCookies(_event: React.SyntheticEvent<HTMLButtonElement>) {
+    this.setState({
+      hasAgreedToCookies: true
+    })
+    this.props.cookies!.set(hasAgreedToCookiesKey, true)
+  }
+
+  render() {
+    // show banner if they haven't clicked okay, otherwise show nothing
+    return (
+      this.state.hasAgreedToCookies ?
+        false
+        :
+        (
+          <div className="cookiesBanner">
+            This site uses Cookies to enhance your experience and to analyze our traffic. Using Synapse means that you agree with our cookie policy.
```

Comment:
Replace "Synapse" with "this website" for portals (since it might be unclear that this is a Synapse portal)?